sys/net/gnrc/network_layer/ipv6/nc/gnrc_ipv6_nc.c

```diff
@@ -163,8 +163,16 @@ void gnrc_ipv6_nc_remove(kernel_pid_t iface, const ipv6_addr_t *ipv6_addr)
 #endif
 #ifdef MODULE_GNRC_SIXLOWPAN_ND_ROUTER
         xtimer_remove(&entry->type_timeout);
+
+        gnrc_ipv6_netif_t *if_entry = gnrc_ipv6_netif_get(iface);
+
+        if (if_entry != NULL &&
+              if_entry->rtr_adv_msg.content.ptr == (char *) entry) {
+            // cancel timer set by gnrc_ndp_rtr_sol_handle
+            xtimer_remove(&if_entry->rtr_adv_timer);
+        }
 #endif
-#if defined(MODULE_GNRC_NDP_ROUTER) || defined(MODULE_GNRC_SIXLOWPAN_ND_BORDER_ROUTER)
+#if defined(MODULE_GNRC_NDP_ROUTER) || defined(MODULE_GNRC_SIXLOWPAN_ND_BORDER_ROUTER) || defined(MODULE_GNRC_SIXLOWPAN_ND_ROUTER)
```

Comment:
`rtr_adv_timer` is not defined in `gnrc_ipv6_nc_t` for `MODULE_GNRC_SIXLOWPAN_ND_ROUTER`